src/cdk-experimental/ui-patterns/behaviors/list-selection/list-selection.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {signal, Signal, WritableSignal} from '@angular/core';
+import {ListNavigation, ListNavigationItem} from '../list-navigation/list-navigation';
+
+/** The required properties for selection items. */
+export interface ListSelectionItem extends ListNavigationItem {
+  /** A unique identifier for the item. */
+  id: Signal<string>;
+
+  /** Whether an item is disabled. */
+  disabled: Signal<boolean>;
+}
+
+/** The required inputs for list selection. */
+export interface ListSelectionInputs<T extends ListSelectionItem> {
+  /** The items in the list. */
+  items: Signal<T[]>;
+
+  /** Whether multiple items in the list can be selected at once. */
+  multiselectable: Signal<boolean>;
+
+  /** The ids of the current selected items. */
+  selectedIds: WritableSignal<string[]>;
+
+  /** The selection strategy used by the list. */
+  selectionMode: Signal<'follow' | 'explicit'>;
+}
+
+/** Controls selection for a list of items. */
+export class ListSelection<T extends ListSelectionItem> {
+  /** The id of the previous selected item. */
+  anchorId = signal<string | null>(null);
```

Comment:
```suggestion
  anchorId = signal<string | undefined>();
```
It's more idiomatic to leave uninitialized things as `undefined` instead of explicitly setting them to `null`. The parts of Angular that explicitly initialize things to `null` are leftovers from when the codebase supported Dart (which doesn't have `undefined`, only `null`)

Reply:
Sure, I went with null because it is shorter and you still need to specify the initial value as undefined

```ts
anchorId = signal<string | null>(null);
vs
anchorId = signal<string | undefined>(undefined);
```